corrected priorityClassName used for static pod (#47055)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/node-pressure-eviction.md

If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
and want to avoid having it evicted under resource pressure, set the
`priorityClassName` field for that Pod directly. 

For example, the following configuration sets the priorityClassName for a static pod as `system-cluster-critical`. 

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: static-web
  labels:
    role: myrole
spec:
  containers:
    - name: web
      image: nginx
      ports:
        - name: web
          containerPort: 80
          protocol: TCP

>   priorityClassName: system-cluster-critical

Author: Changed priority class

If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
and want to avoid having it evicted under resource pressure, set the
`priorityClassName` field for that Pod directly. 

For example, the following configuration sets the priorityClassName for a static pod as `xyz-priority`. 

```yaml
kubectl get priorityclass
NAME                      VALUE        GLOBAL-DEFAULT   AGE
system-cluster-critical   2000000000   false            2m2s
system-node-critical      2000001000   false            2m2s

> xyz-priority              1000000      false            47s

Reviewer: (nit) We prefer to [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output)

Author: Fixed nit

```

```yaml

> apiVersion: v1

Reviewer: ```suggestion
# This is a definition for a static Pod
# You would define this directly for the kubelet on a particular node
# (and not via kubectl)
apiVersion: v1
```

kind: Pod
metadata:
  name: static-web
  labels:
    role: myrole
spec:
  containers:
    - name: web
      image: nginx
      ports:
        - name: web
          containerPort: 80
          protocol: TCP

>   priorityClassName: xyz-priority

Reviewer: This will be fine with one extra caveat: if your static Pod could ever start on a cluster that doesn't yet have a running control plane, the kubelet won't be able to resolve the PriorityClass.

This is important because running the control plane is likely the most common use for static pods, and people probably do want to protect those from eviction.

Reviewer: The work-around is to _also_ define `priority` for any static Pod that might need it.

Author: When I set priorityClass only, the priority is displayed in the description of the pod.  Are you stating to set the priority number higher than any others, for this example?

Author: Only setting priority fails, and is the original reason for this issue.

Reviewer: Sure. You have to set **both** for any static Pod that might start before there's a healthy control plane. The existing docs are wrong; I stand by https://github.com/kubernetes/website/pull/47055#discussion_r1663601731 though.

(and, TBH, the way the admission control mechanism works is unhelpful, but we need to document how K8s actually works).

If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
and want to avoid having it evicted under resource pressure, set the
`priorityClassName` field for that Pod directly. 

For example, the following configuration sets the priorityClassName for a static pod as `xyz-priority`. 

> ```yaml

Reviewer 2: The snippet is not a YAML.

Author: fixed nit.

If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
and want to avoid having it evicted under resource pressure, set the
`priorityClassName` field for that Pod directly. 

> {{<note>}}

Reviewer: (nit)
```suggestion
{{< note >}}
```

> Kubelet considers all static pods as critical regardless of their PodSpec.PriorityClassName. Static pods pass kubelet admission even if a node does not have enough resources. Users must ensure that they account for resources when creating static pods.

Reviewer: How about:
```suggestion
The kubelet considers all static pods as critical regardless of their `.spec.priorityClassName`.
Static pods also pass kubelet admission even if a node does not have enough resources; the kubelet may
instead evict another Pod (potentially even evicting a different Pod at a higher priority, if there is no better
match).

You must ensure that you account for resource use when defining static pods on existing nodes.
```

> {{</note>}}

Reviewer: (nit)
```suggestion
{{< /note >}}
```

> For example, the following configuration sets the priorityClassName for a static pod as `xyz-priority`. 

Reviewer: ```suggestion
For example, you might follow the following steps to set the `priorityClassName` for a static pod as `xyz-priority`. 
```
